### PR TITLE
Added a way to examine submodule changes

### PR DESCRIPTION
**Pull request by mandeep**

### CODE REVIEW

**Comment by msarahan** [comment]
Nice. Looking forward to trying this tomorrow.

File: tests/conftest.py [hunk 1]
```diff
@@ -64,6 +64,41 @@ def testing_git_repo(testing_workdir, request):
...
+                           'https://github.com/conda-forge/conda-build-feedstock.git'])
+
+    subprocess.check_call(['git', 'add', '.'])
+    subprocess.check_call(['git', 'commit', '-m', 'Added submodules'])
+
+    os.chdir('conda-feedstock')
+    subprocess.check_call(['git', 'checkout', '4648194ca029603b90de22e16b59949b5f68d2d5'])
+    os.chdir(testing_workdir)
+
+    subprocess.check_call(['git', 'mv', 'conda-build-feedstock', 'cb3-feedstock'])
```
Comment (msarahan) [inline tests/conftest.py]: Still not sure why you don't have a commit at the end here.  Can you explain what you see as the workflow that gets us to this state?

I think that the state that this fixture creates represents some state on a developer's machine, just prior to a commit.  The aggregate should not actually ever be in this state when c3i runs on it - it should always be clean, with everything committed.

Comment (mandeep) [inline tests/conftest.py]: I wasn't thinking straight. I pushed a new commit that should align with the actual use case.

File: tests/test_compute_build_graph.py [hunk 2]
```diff
@@ -360,3 +360,14 @@ def test_version_matching(testing_conda_resolve):
...
+
+def test_submodules(testing_submodule_commit):
+    assert 'conda-feedstock' in compute_build_graph.git_changed_submodules('.')
+    assert 'cb3-feedstock' in compute_build_graph.git_renamed_folders('.')
+
+
+def test_new_submodules(testing_new_submodules):
+    new_submodules = compute_build_graph.git_new_submodules('.')
+    assert 'conda-env-feedstock' in new_submodules
+    assert 'conda-verify' not in new_submodules
```
Comment (msarahan) [inline tests/test_compute_build_graph.py]: What exactly is this a test of?  does the conda-verify repo not contain a recipe?  Please add some comments here and in your fixture creation.

Comment (mandeep) [inline tests/test_compute_build_graph.py]: Right, conda-verify doesn't have a recipe so I'm testing that it isn't picked up as a submodule to add. I'll add docstrings to the tests to make this clearer.

**Comment by msarahan** [comment]
Nice work here, Mandeep.  I'll look forward to using it on real packages when I get back.

**Pull request merged**